py/reporter_service.py

```diff
@@ -118,40 +94,14 @@ def parse_trace(self, post):
         return json.loads(params['json'][0])
     raise Exception('No json provided')
 
+
   #report some segments to the datastore
   def report(self, trace, partial):
     uuid = trace['uuid']
-    #we buffer this point for now until we get more data
-    if not partial and trace['sq_distance'] < square_distance_cutoff \
-                   and trace['elapsed_time'] < os.environ.get('MIN_ELAPSED_TIME', 60):
-      thread_local.cache.set(uuid, pickle.dumps(trace), ex=os.environ.get('PARTIAL_EXPIRY', 300))
-      return None
-
     #ask valhalla to give back OSMLR segments along this trace
     result = thread_local.segment_matcher.Match(json.dumps(trace, separators=(',', ':')))
     segments = json.loads(result)
 
-    #TODO: there are a couple of problems below:
-    #if a probe just sits in the same place and continues getting a partial we'll do matching every time
-    #if a probe drops off before hitting the threshhold we will expire that data without ever matching it
-
-    #partials we dont keep around they are done from here
-    if partial:
-      thread_local.cache.delete(uuid)
-    #there is some partially traversed segment we want to use again
-    elif len(segments['segments']) and segments['segments'][-1]['start_time'] >= 0 and segments['segments'][-1]['end_time'] < 0:
-      trace['trace'] = trace['trace'][segments['segments'][-1]['begin_shape_index']:]
-      sq_distance, elapsed_time = sum_difference(trace['trace'])
-      trace['sq_distance'] = sq_distance
-      trace['elapsed_time'] = elapsed_time
-      thread_local.cache.set(uuid, pickle.dumps(trace), ex=os.environ.get('PARTIAL_EXPIRY', 300))
-    #either no segments or the last one was complete either way keep the last point for the next try
-    else:
-      trace['trace'] = trace['trace'][-1:]
-      trace['sq_distance'] = 0
-      trace['elapsed_time'] = 0
-      thread_local.cache.set(uuid, pickle.dumps(trace), ex=os.environ.get('PARTIAL_EXPIRY', 300))
-
     #clean out the unuseful partial segments
     segments['segments'] = [ seg for seg in segments['segments'] if seg['length'] > 0 ]
```

Comment:
we'll also need to return the last partial valid segment or at least the `begin_shape_index` of that. if that doesnt exist then we could return a value to signify to flink that that we used the whole shape